[update] Install self-hosted TimescaleDB on macOS systems (#1786)
timescale/docs: install/installation-macos.md

psql -U postgres -h localhost -d tsdb
```

## Uninstalling TimescaleDB

> You can uninstall TimescaleDB installed using Homebrew or MacPorts without
> having to uninstall PostgreSQL.

Reviewer: Let's make this more active:

```suggestion
If you installed TimescaleDB with Homebrew or MacPorts, you can uninstall 
it without having to uninstall PostgreSQL.
```

<procedure>

### Uninstalling TimescaleDB using Homebrew

> To uninstall TimescaleDB installed using Homebrew

Reviewer: ```suggestion
```

> 1.  Remove the TimescaleDB extension:

Reviewer: ```suggestion
1.  At the `psql` prompt, remove the TimescaleDB extension:
```

>     ```bash

Reviewer: ```suggestion
    ```sql
```

>     tsdb=# DROP EXTENSION timescaledb;

Reviewer: ```suggestion
    DROP EXTENSION timescaledb;
```

    ```

> 1.  Remove `timescaledb` from `shared_preload_libraries` in the `postgresql.conf`:

Reviewer: ```suggestion
1.  At the command prompt, remove `timescaledb` from `shared_preload_libraries` 
    in the `postgresql.conf` configuration file:
```

    ```bash
    nano /opt/homebrew/var/postgresql@14/postgresql.conf
    shared_preload_libraries = ''
    ```

1.  Save the changes to the `postgresql.conf` file.

1.  Restart PostgreSQL:

    ```bash
    brew services restart postgresql
    ```

1.  Check that the TimescaleDB extension is uninstalled by using the `\dx`
    command at the `psql` prompt. Output is similar to:

    ```sql
    tsdb-# \dx
                                          List of installed extensions
        Name     | Version |   Schema   |                            Description                            
    -------------+---------+------------+-------------------------------------------------------------------
     plpgsql     | 1.0     | pg_catalog | PL/pgSQL procedural language
    (1 row) 
    ```

1.  Uninstall TimescaleDB:

    ```bash

>     brew uninstall timescaldb

Reviewer: ```suggestion
    brew uninstall timescaledb
```

    ```

1.  Remove all the dependencies and related files:

    ```bash
    brew remove timescaledb
    ```

</procedure>

<procedure>

### Uninstalling TimescaleDB using MacPorts

> To uninstall TimescaleDB installed using Macports:

Reviewer: ```suggestion
```

> 1.  Remove the TimescaleDB extension:

Reviewer: ```suggestion
1.  At the `psql` prompt, remove the TimescaleDB extension:
```

>     ```bash

Reviewer: ```suggestion
    ```sql
```

>     tsdb=# DROP EXTENSION timescaledb;

Reviewer: ```suggestion
    DROP EXTENSION timescaledb;
```

    ```

> 1.  Remove `timescaledb` from `shared_preload_libraries` in the `postgresql.conf`:

Reviewer: ```suggestion
1.  At the command prompt, remove `timescaledb` from `shared_preload_libraries` 
    in the `postgresql.conf` configuration file:
```

    ```bash
    nano /opt/homebrew/var/postgresql@14/postgresql.conf
    shared_preload_libraries = ''
    ```

1.  Save the changes to the `postgresql.conf` file.

1.  Restart PostgreSQL:

    ```bash
    port reload postgresql
    ```

1.  Check that the TimescaleDB extension is uninstalled by using the `\dx`
    command at the `psql` prompt. Output is similar to:

    ```sql
    tsdb-# \dx
                                          List of installed extensions
        Name     | Version |   Schema   |                            Description                            
    -------------+---------+------------+-------------------------------------------------------------------
     plpgsql     | 1.0     | pg_catalog | PL/pgSQL procedural language
    (1 row) 
    ```

1.  Uninstall TimescaleDB and the related dependencies:

    ```bash

>     port uninstall timescaldb --follow-dependencies

Reviewer: ```suggestion
    port uninstall timescaledb --follow-dependencies
```

    ```

</procedure>

> ## Where to next

Reviewer: This feels a little weird coming after uninstalling now ... 🤔

Author: @Loquacity  thought so too....

Reviewer: I wonder if it's better to have uninstalling in its own section somewhere, rather than as part of install. We could stick it down in the How Tos. Or even make it a troubleshooting section 🤔 @charislam do you have any thoughts?

Reviewer: OK, so let's create an 'Uninstalling TimescaleDB" section under HowTos, then 👍

Author: cool!!! let me commit the changes then!!!
